Update s3.md

LuciMoore opened the pull request:
(1) updated intro text to indicate how to gain access to a bucket and (2) cleaned up s3 commands section to only contain info that isn't already included in central documentation. I also removed:

  - If your job runs out of time before the data upload is complete, use `s3cmd sync --recursive --no-check-md5` to restart the upload process where it left off and assure  everything makes it. Run this even if the sync says complete to validate

because we should encourage people to use platforms like Globus for this instead

rosemccollum reviewed (changes requested):
Please put the commands back. 
Also can you clarify the note about the getx500 not including yourself (get_x500 won't work if you don't already have access to the bucket) and maybe move the note about how you can't give yourself access to a bucket down to the policy setting section?

LuciMoore commented:
@rosemccollum I made the following changes:

- changed the title of "Setting a policy on s3" to "Updating S3 Policy & Bucket Access" to make the purpose of the section more clear for folks who are not familiar with what setting a bucket policy means. now that the header is more clear, I moved the note about how you can't give yourself access to a bucket down to this section
- added more introductory information in the s3 usage section to explain the behavior of the s3 and why/how it's different from interacting with a filesystem
- added back the s3 commands
-clarified that get_x500 won't work if you don't already have access to the bucket
-cleaned up the s3 config section a bit for clarity to state purpose
